test/RequestChannelTest.cpp

```diff
@@ -88,11 +88,299 @@ TEST(RequestChannelTest, RequestOnDisconnectedClient) {
   ASSERT(did_call_on_error);
 }
 
-// TODO complete from requester, responder continues
-// TODO complete from responder, requester continues
-// TODO cancel from requester, shuts down
-// TODO flow control from requester to responder
-// TODO flow control from responder to requester
-// TODO failure on responder, requester sees
-// TODO failure on request, requester sees
-// TODO failure from requester ... what happens?
+class TestChannelResponder : public rsocket::RSocketResponder {
+ public:
+  TestChannelResponder(
+      int64_t rangeEnd = 10,
+      int64_t initialSubReq = credits::kNoFlowControl)
+      : rangeEnd_{rangeEnd},
+        testSubscriber_{TestSubscriber<std::string>::create(initialSubReq)} {}
+
+  yarpl::Reference<Flowable<rsocket::Payload>> handleRequestChannel(
+      rsocket::Payload initialPayload,
+      yarpl::Reference<Flowable<rsocket::Payload>> requestStream,
+      rsocket::StreamId) override {
+    // add initial payload to testSubscriber values list
+    testSubscriber_->onNext(initialPayload.moveDataToString());
+
+    requestStream->map([](auto p) { return p.moveDataToString(); })
```

Comment:
Looks like travis is pointing at this line as where the leak happens for the `cancel` test. I would test copying the payload instead of moving it and see if it fails still.